app/facades/video_facade.rb

```diff
@@ -0,0 +1,39 @@
+class VideoFacade
+  def initialize(vimeo_id)
+    @vimeo_id = vimeo_id
+  end
+
+  def title
+    video_data[:name]
+  end
+
+  def description
+    video_data[:description]
+  end
+
+  def length
+    Time.at(video_data[:duration]).utc.strftime("%H:%M:%S")
+  end
+
+  def video_url
+    video_data[:link]
+  end
+
+  def thumbnail_url
+    video_data[:pictures][:sizes][3][:link_with_play_button]
+  end
+
+  def embed_code
+    video_data[:embed][:html]
+  end
+
+  private
+
+  def service
+    @_service ||= VimeoService.new
+  end
+
+  def video_data
+    @_video_data ||= service.get_video(@vimeo_id)
```

Comment:
Taking into account how most of the VideoFacade methods depend upon the return value of `service.get_video`, I'm curious about how gracefully nil values are handled. For example, would line 27 (`video_data[:embed][:html]`) error out if `video_data[:embed]` is nil?

If that method doesn't return the expected video, could it return something else? I think there are several ways you could handle this edge case, and you might be doing so already, just something to think about.